iac-extensions/docker/src/test/java/org/sonar/iac/docker/symbols/ArgumentResolutionTest.java

```diff
@@ -166,6 +172,65 @@ void shouldNotDeadLoopWhenResolvingSelfAssignedVariable() {
     assertThat(resolution.status()).isEqualTo(UNRESOLVED);
   }
 
+  @ParameterizedTest
+  @MethodSource
+  void shouldCorrectlyHandleQuotes(String input, String expectedSecondArgument) {
+    File file = parseFileAndAnalyzeSymbols(input);
+
+    List<ArgumentResolution> argumentResolutions = CheckUtils.resolveInstructionArguments(TreeUtils.firstDescendant(file, RunInstruction.class).get());
+    ArgumentResolution stringArgument = argumentResolutions.get(1);
+
+    assertThat(stringArgument.value()).isEqualTo(expectedSecondArgument);
+  }
+
+  static Stream<Arguments> shouldCorrectlyHandleQuotes() {
+    return Stream.of(
+      // argument of `echo` is StringLiteral
+      "RUN echo foo#foo",
+      "RUN echo \"foo\"#\"foo\"",
+      "RUN echo 'foo'#'foo'",
+      "RUN echo '\"foo\"'#'\"foo\"'",
+      "RUN echo \"\"foo\"\"#\"\"foo\"\"",
+      "RUN echo \"'foo'\"#\"'foo'\"",
+      // argument of `echo` is ExpandableStringLiteral
+      "RUN echo $FOO#foo",
+      "RUN echo \"$FOO\"#\"foo\"",
+      // single-quoted variables are not substituted by Docker, so we shouldn't resolve them
+      "RUN echo '$FOO'#'$FOO'",
+      "RUN echo \"\"$FOO\"\"#\"\"foo\"\"",
+      "RUN echo ${FOO}#foo",
+      "RUN echo \"${FOO}\"#\"foo\"",
+      "RUN echo '${FOO}'#'${FOO}'",
+      "RUN echo \"\"${FOO}\"\"#\"\"foo\"\"",
+      // in ExecForm all arguments in an array are ExpandableStringLiterals
+      "RUN [\"echo\", \"$FOO\"]#foo",
+      "RUN [\"echo\", \"'$FOO'\"]#'foo'",
+      "RUN [\"echo\", \"${FOO}\"]#foo",
+      "RUN [\"echo\", \"'${FOO}'\"]#'foo'",
+      // in ExecForm quotes have to be escaped and we are not un-escaping them
```

Comment:
Does make it sense to unescape quotes in the case of ExecForm? Then we will not need to care about where the value comes from. 🤔

Reply:
I think it would make sense to unescape them during argument resolution, because then we will have access to actual string values that will be passed to the launched process. I'll create a ticket for this topic.